socketserver/server/publisher.go

```diff
@@ -174,8 +199,25 @@ func HTTPBackendCachedPublish(w http.ResponseWriter, r *http.Request) {
 		saveLastMessage(cmd, channel, expires, json, deleteMode)
 	}
 	CachedLSMLock.Unlock()
-	count = PublishToMultiple(channels, msg)
 
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go rl.Run()
+	go func() {
+		count = PublishToMultiple(channels, msg, rl)
+		wg.Done()
+		rl.Close()
+	}()
+	ch := make(chan struct{})
```

Comment:
this should probably have a comment:

    // Time out and return a count of -1 if it takes more than 3 seconds to publish the message.

The purpose of the code isn't obvious.